backend/kemomimi_api/src/routes/public_items/index.rs

```diff
@@ -125,7 +125,31 @@ impl PublicItems for ApiImpl {
     where
         'life0: 'async_trait,
     {
-        todo!()
+        let public_item_id = Uuid::now_v7();
+
+        let query = r#"
+            INSERT INTO public_item (
+                public_item_id,
+                name,
+                category_id,
+                cost,
+                purchase_date,
+                expiration_date,
+                is_remaining,
+                remarks
+            )
+            VALUES (
+                $1,
+                $2,
+                $3,
+                $4,
+                $5,
+                $6,
+                $7,
+                $8
+            )
+            RETURNING public_item_id;
+        "#;
```

Comment:
型とテーブルをチェックするマクロ`query_as!`を使いたいので、クエリ文字列は直接書いてください。(もしくはマクロで置換)
```suggestion
        let public_item_id = query_as!(
            Uuid,
            r#"
            INSERT INTO public_item (
                public_item_id,
                name,
                // category_id,
                cost,
                purchase_date,
                expiration_date,
                is_remaining,
                remarks
            )
            VALUES (
                $1,
                $2,
                $3,
                $4,
                $5,
                $6,
                $7,
                $8
            )
            RETURNING public_item_id;
        "#
        );
```